Sources/SpeziStudy/Study Manager/StudyManager.swift

```diff
@@ -415,8 +427,12 @@ extension StudyManager {
     /// - if `X = Y`: have no effect;
     /// - if `X < Y`: update the study enrollment, as if ``informAboutStudies(_:)`` was called with the new study revision;
     /// - if `X > Y`: throw an error.
+    ///
+    /// - parameter studyBundle: The `StudyBundle` to enroll into.
+    /// - parameter enrollmentDate: The `Date` relative to which the enrollment should be registered.
+    ///     This defaults to the current date, but you can override this to specify a date in the past if you re-enroll a user which was already enrolled before.
     @MainActor
-    public func enroll(in studyBundle: StudyBundle) async throws {
+    public func enroll(in studyBundle: StudyBundle, enrollmentDate: Date = .now) async throws {
```

Comment:
How do we handle task completion for the study? Would the app then be responsible to complete any tasks that a user has already completed in the past or would that be handled by Spezi Study? Spezi Study currently only stores the data locally in Swift Data and if the app was deleted it would be gone if not synced to a server? I suspect it is the responsibility of the app as we would get this information from the server if we store it there and then re-complete all tasks? 

In the future we would solve this by using the Swift Data Backend Storage mechanism that would allow us to sync the completion elements from there?